include/deal.II/base/mg_level_object.h

```diff
@@ -207,7 +220,7 @@ MGLevelObject<Object>::resize(const unsigned int new_minlevel,
 
   minlevel = new_minlevel;
   for (unsigned int i = 0; i < new_maxlevel - new_minlevel + 1; ++i)
-    objects.push_back(std::make_shared<Object>());
+    objects.push_back(std::make_shared<Object>(std::forward<Args>(args)...));
```

Comment:
You could avoid one copy if you used `emplace_back`.

Reply:
How is that? The `std::shared_ptr` is moved for both `push_back` and `emplace_back` (see https://en.cppreference.com/w/cpp/container/vector/push_back) or are you referring to something else?

Reply:
I don't have any preferences here.

Reply:
Of course, we could also use `std::fill_n` with `std::back_inserter` here if we really want to make the intent clear that all of them are the same. But this solution is fine with me.

Reply:
Ah, nevermind. We want separate `std::shared_ptr`s od course.

Reply:
I don't matter much to me. We've tried to move away from `push_back` and use `emplace_back` instead, but you're right that it doesn't matter here because there's an overload of `push_back` that takes an rvalue reference.